### PR TITLE
pn_bgp (New Module)

### PR DESCRIPTION
**Pull request by amitsi**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

BGP Module
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/etc/ansible/pluribus-ansible/ansible/library']
```
##### SUMMARY

```

```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @amitsi for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by amitsi** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @amitsi for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: network/netvisor/pn_vrouterbgp.py [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+            pn_clipassword=dict(required=False, type='str', no_log=True),
+            pn_cliswitch=dict(required=False, type='str', default='local'),
+            pn_command=dict(required=True, type='str',
+                            choices=['vrouter-bgp-add', 'vrouter-bgp-remove',
+                                     'vrouter-bgp-modify']),
+            pn_vrouter_name=dict(required=True, type='str'),
+            pn_neighbor=dict(type='str'),
+            pn_remote_as=dict(type='str'),
+            pn_next_hop_self=dict(type='bool'),
+            pn_password=dict(type='str'),
```
Comment (gundalow) [inline network/netvisor/pn_vrouterbgp.py]: All password (and any other sensitive data) options need

`no_log=True`

**Comment by ashxjain** [comment]
The previous push was accidentally done with --amend, because of which it had @privateip 's changes. Created a clean branch locally and then pushed to Github with --force to overwrite the existing branch (pn_ospf). This is the latest clean commit: ccf299d2eef53c8f66732ec740017608c4f62be1

File: network/netvisor/pn_vrouterbgp.py [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+
+- name: remove vrouter-bgp
+  pn_vrouterbgp:
+    state: 'absent'
+    pn_name: 'ansible-vrouter'
+"""
+
+RETURN = """
+command:
+  description: The CLI command run on the target node(s).
```
Comment (gundalow) [inline network/netvisor/pn_vrouterbgp.py:145]: Missing `returned:` and `type:`

File: network/netvisor/pn_vrouterbgp.py [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+            pn_ebgp=dict(type='int'),
+            pn_prefix_listin=dict(type='str'),
+            pn_prefix_listout=dict(type='str'),
+            pn_route_reflector=dict(type='bool'),
+            pn_override_capability=dict(type='bool'),
+            pn_soft_reconfig=dict(type='bool'),
+            pn_max_prefix=dict(type='int'),
+            pn_max_prefix_warn=dict(type='bool'),
+            pn_bfd=dict(type='bool'),
+            pn_multiprotocol=dict(type='bool',
```
Comment (gundalow) [inline network/netvisor/pn_vrouterbgp.py]: This isn't a `bool` if it accepts choices=['ipv4-unicast', 'ipv6-unicast']

I think you mean `type:'str'`

**Review by gundalow — CHANGES_REQUESTED** [review]

File: network/netvisor/pn_vrouterbgp.py (hunk 3 above)
Comment (gundalow) [reply network/netvisor/pn_vrouterbgp.py]: Please check this then I can merge

**Comment by gundalow** [comment]
needs_revision

File: network/netvisor/pn_vrouterbgp.py (hunk 2 above)
Comment (ashxjain) [reply network/netvisor/pn_vrouterbgp.py:145]: That's just the description for the command. `returned:` and `type:` are mentioned below.

**Comment by ashxjain** [comment]
ready_for_review

File: network/netvisor/pn_vrouterbgp.py (hunk 2 above)
Comment (gundalow) [reply network/netvisor/pn_vrouterbgp.py:145]: Yup, you are correct. PR blindness :)

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merged, thanks!

**Comment by ashxjain** [comment]
Thanks a lot @gundalow !

**Pull request merged**